### PR TITLE
Fix compile errors and warnings

### PR DESCRIPTION
**Pull request by aeisenberg**
Here are a few behaviour preserving changes that I made that helps avoid some compiler errors and warnings.

- Updated launch config for mocha (tests are failing, but at least it launches)
- Updated versions in package-lock.json
- Updated `@types/vscode` to latest
- Removed two `onCommand` triggers that are now deprecated
- Added `const vscode: VsCodeApi` to `global.d.ts` to get properly typed references to `vscode` instance in the webview.
- Converted `return` to `return undefined` where appropriate to make compiler happy.
- Also, a handful of other errors that the compiler was complaining about.

### CODE REVIEW

**Comment by aeisenberg** [comment]
@EasyRhinoMSFT, would you be able to take a look at this?

File: .vscode/launch.json [hunk 1]
```diff
@@ -16,9 +16,8 @@
             "name": "Launch Mocha",
             "type": "node",
             "request": "launch",
-            "program": "${workspaceRoot}/node_modules/mocha/bin/mocha",          
-            "args": ["--color", "--inspect-brk=9229"], // "${file}"
-            "port": 9229,
+            "program": "${workspaceRoot}/node_modules/.bin/mocha",
+            "args": ["--color", "--inspect-brk=9229", "${file}"], //
```
Comment (EasyRhinoMSFT) [inline .vscode/launch.json:20]: >// [](http://example.com/codeflow?start=66&length=2)

Remove?

**Review by EasyRhinoMSFT — APPROVED** [review]
:shipit:

File: .vscode/launch.json (hunk 1 above)
Comment (aeisenberg) [reply .vscode/launch.json:20]:
```suggestion
            "args": ["--color", "--inspect-brk=9229", "${file}"],
```

**Comment by aeisenberg** [comment]
Thanks for the review!

**Pull request merged**